### PR TITLE
Fix/Remove storage/import/export bus IItemHandler caching

### LINKED ISSUE #2982: [1.12] Storage Bus not updating
When a storage bus is used together (In this example) with a phantomface it fails to update the contents of the inventory when the phantomface is rebound to a different inventory.

I am pretty sure this is an old bug, not sure if something was decided regarding this.

### PR DESCRIPTION
**Pull request by fscan**
For StorageBus - like the old version - a hash made of TileEntity, IItemHandler and number of slots is built.

Fixes #2982

### CODE REVIEW

File: src/main/java/appeng/parts/misc/PartStorageBus.java [hunk 1]
```diff
@@ -402,6 +403,21 @@ private void resetCache()
...
 	}
 
+	private int createHandlerHash( TileEntity target )
+	{
+		final EnumFacing targetSide = this.getSide().getFacing().getOpposite();
+		if( target.hasCapability( Capabilities.STORAGE_MONITORABLE_ACCESSOR, targetSide ) )
+		{
+			return 0;
+		}
+		IItemHandler handler = target.getCapability( CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, targetSide );
```
Comment (orod-org) [inline src/main/java/appeng/parts/misc/PartStorageBus.java]: MAJOR Rename "handler" which hides the field declared at line 115. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)

**Comment by orod-org** [comment]
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 4 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Platform.java#L604](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/50e57a94d579b35e7d7cd4b8ccfbfe99a902280a/src/main/java/appeng/util/Platform.java#L604): Remove this useless assignment to local variable "itemStack". [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1854)
1. ![MAJOR][MAJOR] [Platform.java#L649](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/50e57a94d579b35e7d7cd4b8ccfbfe99a902280a/src/main/java/appeng/util/Platform.java#L649): Remove this useless assignment to local variable "itemStack". [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1854)
1. ![MAJOR][MAJOR] [Platform.java#L1663](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/50e57a94d579b35e7d7cd4b8ccfbfe99a902280a/src/main/java/appeng/util/Platform.java#L1663): This block of commented-out lines of code should be removed. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MINOR][MINOR] [Platform.java#L461](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/50e57a94d579b35e7d7cd4b8ccfbfe99a902280a/src/main/java/appeng/util/Platform.java#L461): Remove this use of "getDrops"; it is deprecated. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by orod-org** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Platform.java#L604](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/8f0618da08b883986123c28bba7b93bc7499b0d9/src/main/java/appeng/util/Platform.java#L604): Remove this useless assignment to local variable "itemStack". [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1854)
1. ![MAJOR][MAJOR] [Platform.java#L649](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/8f0618da08b883986123c28bba7b93bc7499b0d9/src/main/java/appeng/util/Platform.java#L649): Remove this useless assignment to local variable "itemStack". [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1854)
1. ![MAJOR][MAJOR] [Platform.java#L1663](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/8f0618da08b883986123c28bba7b93bc7499b0d9/src/main/java/appeng/util/Platform.java#L1663): This block of commented-out lines of code should be removed. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MINOR][MINOR] [Platform.java#L461](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/8f0618da08b883986123c28bba7b93bc7499b0d9/src/main/java/appeng/util/Platform.java#L461): Remove this use of "getDrops"; it is deprecated. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by fscan** [comment]
fixes #2982

**Comment by xsun2001** [comment]
Hey guys.
I am merging the patches of higher version to 1.7.10 recently. But I cannot cherry-pick this PR easily because of the lack of IItemHandler and Capability system. I would like to know if the merger of this branch can help reduce some potential bugs in 1.7.10. I also want to know how to merge this branch without IItemHandler. Can I call `net.minecraft.inventory.IInventory#getStackInSlot` for every available index to get everything in it?

**Pull request merged**